MTA-6893 Document Hub OIDC authentication (#382)
migtools/mta-documentation: docs/topics/mta-install/con_ldap-token-lifecycle.adoc

:_newdoc-version: 2.18.5
:_template-generated: 2025-07-31
:_mod-docs-content-type: CONCEPT

[id="ldap-token-lifecycle_{context}"]
= LDAP token lifecycle

[role="_abstract"]

> {ProductFirstRef} Hub manages token lifecycle in a Lightweight Directory Access Protocol (LDAP)-based authentication to account for changes in membership, user permission, and user account status.

Reviewer: ```suggestion
The {ProductFirstRef} Hub manages the token lifecycle for LDAP-based authentication to account for changes in membership, user permissions, and user account status.
```

Abbreviations (Red Hat SSG): The Red Hat SSG explicitly dictates not to expand the abbreviation LDAP on its first use.

Missing Articles (IBM Style): Include the definite article "the" before "token lifecycle".

Grammar/Flow: The phrase "in a LDAP-based authentication" is awkward. Changing it to "for LDAP-based authentication" flows better. Also, "permission" should be pluralized to match the other items in the list.

Author: The SSG expands LDAP in the LDAP noun section: https://redhat-documentation.github.io/supplementary-style-guide/#ldap

It also sights the expanded form as the incorrect form. So, it's a little unclear to me what is expected here.

> A user authenticated by LDAP gets an access token that has a default lifespan of 5 minutes. When the access token expires, the authentication server checks for a cached LDAP identity. If the LDAP identity is cached and not expired, the Hub OIDC provider issues a new token with the cached scopes for the user. If the LDAP identity expires or is not cached, the Hub OIDC provider reauthenticates the user by connecting with the Directory Server (DS).

Reviewer: ```suggestion
A user authenticated by LDAP receives an access token that has a default lifespan of 5 minutes. When the access token expires, the authentication server checks for a cached LDAP identity. If the LDAP identity is cached and not expired, the Hub OpenID Connect (OIDC) provider issues a new token with the cached scopes for the user. If the LDAP identity expires or is not cached, the Hub OIDC provider reauthenticates the user by connecting to the Directory Server (DS).
```

The verb "gets" is colloquial. Replacing it with "receives" aligns better with technical tone.

Use "connecting to" rather than "connecting with" for software connections.

> To issue or refresh an access token, the {ProductFullName} Hub connects with the DS by using the stored password, queries the LDAP groups again, maps the LDAP groups to {ProductShortName} roles, and updates the user identity with expiration. The user need not manually reauthenticate to start the token refresh process. 

Reviewer: ```suggestion
To issue or refresh an access token, the {ProductFullName} Hub connects to the DS by using the stored password, queries the LDAP groups again, maps the LDAP groups to {ProductShortName} roles, and updates the user identity with an expiration time. The user does not need to manually reauthenticate to start the token refresh process.
```
